media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -134,48 +168,32 @@ def doRollover(self):
         super().doRollover()
 
-        # Construct a file pattern to match all current gzip'd logs
-        base_file_pattern = self.baseFilename + ".[0-9].gz"
-        # Get a list of the current .gzip'd log files to determine the next extension to use
-        arch_files = glob.iglob(base_file_pattern)
-        # Loop through the archived log files to find the last file number in use
-        lastgz_file_num = '0'  # default value for first iteration (no current gzips)
-        for archFile in arch_files:
-            fn = archFile.split('.')
-            # The gzip'd file "number" is in array element 2
-            lastgz_file_num = fn[2]
-        lastgz_file_int = int(lastgz_file_num)
-        # Construct a gzip file name for the destination by incrementing the last file number
-        # used.  If more than (backupCount) files are already zipped, delete the oldest and rename
-        # the remaining to free up the last number for re-use.  This ensures that the newest gzip
-        # will always be the highest number.
-        nextgz_file_int = lastgz_file_int + 1
-        if nextgz_file_int > self.backupCount:
-            nextgz_file_int = self.backupCount
-            # Since we already have the maximum gzip'd logs, delete the oldest, then rename the
-            # remaining to free up the last slot.
-            oldest_gz_log = self.baseFilename + ".1.gz"
-            os.remove(oldest_gz_log)
-            # Rename the remaining gzip'd logs...
-            arch_files = glob.iglob(base_file_pattern)
-            for archFile in arch_files:
-                fn = archFile.split('.')
-                # The gzip'd file "number" is in array element 2
-                newgz_file_num = int(fn[2]) - 1
-                newgz_file = fn[0] + "." + fn[1] + "." + str(newgz_file_num) + ".gz"
-                os.rename(archFile, newgz_file)
-        else:
-            # haven't hit the maximum backup file count yet, so simply construct the
-            # next gzip file name based on the existing base file name components
-            fn = (self.baseFilename + ".1.gz").split('.')
-
-        nextgz_file_name = fn[0] + "." + fn[1] + "." + str(nextgz_file_int) + ".gz"
+        # Now the rollover has occurred, so we should have an unarchived file
+        # with a {baseFilename}.1 name.  Before compressing it, we need to "roll up"
+        # the existing GZIP'd archives to make room for the new GZIP we'll
+        # create.  (The newest GZIP archive has a ".1.gz" extension.)
+        for gzip_idx in range(self.backupCount, 0, -1):
+            gzip_filename = f'{self.baseFilename}.{gzip_idx}.gz'
+            if os.path.exists(gzip_filename):
+                if (gzip_idx + 1) > self.backupCount:
+                    # if the current GZIP is "rolled up", it will exceed the maximum
+                    # number of backup files to be kept, so remove it
+                    os.remove(gzip_filename)
```

Comment:
Not really that important, but FWIW: I don't think you need this step, because the n-1'th logfile will end up replacing the last logfile, anyway (via `os.rename()`).

Reply:
I get your point...but with the current range for the loop, we either need to delete the file, "check and pass" if we're going to exceed the backupCount (resulting in it getting overwritten), or change the loop starting point so that it's (backupCount - 1).